### PR TITLE
[Mac] Add "New Instance" command to menu item when right-clicking on …

### PR DESCRIPTION
**Pull request by DavidKarlas**
…instance of IDE in Mac Dock
A picture is worth a thousand words:
image

Enough `open -n`

### CODE REVIEW

File: main/src/addins/MacPlatform/MacPlatform.cs [hunk 1]
```diff
@@ -298,6 +299,27 @@ public override Xwt.Toolkit LoadNativeToolkit ()
...
+				return;
+			var menu = new NSMenu ();
+			var newInstanceMenuItem = new NSMenuItem ();
+			newInstanceMenuItem.Title = GettextCatalog.GetString ("New Instance");
+			newInstanceMenuItem.Activated += NewInstanceMenuItem_Activated;
+			menu.AddItem (newInstanceMenuItem);
+			e.DockMenu = menu;
+		}
+
+		void NewInstanceMenuItem_Activated (object sender, EventArgs e)
```
Comment (Therzok) [inline main/src/addins/MacPlatform/MacPlatform.cs]: I guess this can be static.

File: main/src/addins/MacPlatform/MacPlatform.cs [hunk 2]
```diff
@@ -298,6 +299,27 @@ public override Xwt.Toolkit LoadNativeToolkit ()
...
+		{
+			var bundlePath = NSBundle.MainBundle.BundlePath;
+			NSWorkspace.SharedWorkspace.LaunchApplication (NSUrl.FromFilename (bundlePath), NSWorkspaceLaunchOptions.NewInstance, new NSDictionary (), out NSError error);
+			if (error != null)
+				LoggingService.LogError ($"Failed to start new instance: { error.LocalizedDescription}");
```
Comment (Therzok) [inline main/src/addins/MacPlatform/MacPlatform.cs]:
```suggestion
				LoggingService.LogError ($"Failed to start new instance: {error.LocalizedDescription}");
```

File: main/src/addins/MacPlatform/MacPlatform.cs (hunk 1 above)
Comment (Therzok) [inline main/src/addins/MacPlatform/MacPlatform.cs]:
```suggestion
		static void NewInstanceMenuItem_Activated (object sender, EventArgs e)
```

**Comment by mrward** [comment]
This is nice. Just wondering about the label 'New Instance'.

- VS Code uses 'New Window'
- Terminal uses 'New Window'
- Finder uses 'New Finder Window'

**Comment by DavidKarlas** [comment]
@mrward but macOS API says NewInstance, joke aside
All those apps don’t actually spawn another app which has new icon in Dock.
Would likw others input, I don’t care much about name tbh :)

**Comment by slluis** [comment]
"New Instance" is ok, maybe some day we'll add support for "New Window".

**Comment by vancura** [comment]
What about "New Window"? "New Instance" sounds too robotic. @hbons thoughts?

**Comment by vancura** [comment]
Yeah, @Therzok is right, I haven't seen the rest of the comments, sorry (too swamped).

Keep New Instance then. On the end we're making an editor for people who know what an instance is, this is not iPhoto :)

**Comment by hbons** [comment]
Agree with @slluis. If it launches with another dock icon: "New Instance" is ok. Otherwise "New Window".

**Comment by jmatthiesen** [comment]
FYI - I just added a reference in the description to a VSTS item for this, for ease of tracking with release notes.

**Pull request merged**